### PR TITLE
Add Nova editor extension types

### PR DESCRIPTION
**Pull request by apexskier**
This adds types for extensions built for the new editor, [Nova](https://novadocs.panic.com), from @panicinc. It's a custom JavaScriptCore runtime, so the base types are different. However, `@types/node` is really hard to get rid of, so I've created two types for compatibility (https://github.com/apexskier/DefinitelyTyped/blob/nova-editor/types/nova-editor/README.md).

The [typescript language plugin](https://github.com/apexskier/nova-typescript) I'm writing uses these and is how I've been testing.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/nova-editor/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
+// Type definitions for non-npm package nova-editor 1.0
+// Project: https://novadocs.panic.com
+// Definitions by: Cameron Little <https://github.com/apexskier>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.0:
```
Comment (orta) [inline types/nova-editor/index.d.ts:5]:
```suggestion
// Minimum TypeScript Version: 3.0
```

**Comment by apexskier** [comment]
The build isn't actually failing.

**Comment by apexskier** [comment]
@typescript-bot The build isn't failing

**Comment by orta** [comment]
Sorry - new bot, we're working out all the kinks ATM

**Pull request merged**